projects/dsp-ui/src/lib/search/expert-search/expert-search.component.ts

```diff
@@ -0,0 +1,117 @@
+import { Component, EventEmitter, Inject, Input, OnInit, Output } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { KnoraApiConfig } from '@dasch-swiss/dsp-js';
+import { DspApiConfigToken } from '../../core/core.module';
+import { AdvancedSearchParams, AdvancedSearchParamsService } from '../services/advanced-search-params.service';
+
+@Component({
+    selector: 'dsp-expert-search',
+    templateUrl: './expert-search.component.html',
+    styleUrls: ['./expert-search.component.scss']
+})
+export class ExpertSearchComponent implements OnInit {
+
+    /**
+     * @param gravsearchQuery Send the gravsearch query back.
+     */
+    @Output() gravsearchQuery = new EventEmitter<string>();
+
+    /**
+     * @param toggleExpertSearchForm Trigger toggle for extended search form.
+     */
+    @Output() toggleExpertSearchForm = new EventEmitter<boolean>();
+
+    expertSearchForm: FormGroup;
+
+    constructor(
+        @Inject(DspApiConfigToken) private dspApiConfig: KnoraApiConfig,
+        private _searchParamsService: AdvancedSearchParamsService,
+        private fb: FormBuilder,
+        private _route: ActivatedRoute,
+        private _router: Router
+    ) { }
+
+    ngOnInit(): void {
+        this.initForm();
+    }
+
+    /**
+     * @ignore
+     * Initiate the form with predefined Gravsearch query as example.
+     */
+    private initForm() {
```

Comment:
You can also instantiate a `FormControl` and assign to a class member:

https://github.com/dasch-swiss/dsp-ui-lib/blob/b3b1fee1ecc97085f3bdb199c0352c36825b206a/projects/dsp-ui/src/lib/viewer/values/int-value/int-value.component.ts#L41-L55

Then you do not need to access it via the string within your class.

Reply:
done in 0fe3539

Reply:
You can now use `this.queryFormControl.value` instead of `const queryTemplate = this.expertSearchForm.controls['gravsearchquery'].value;`

Reply:
done in 26e2af2